src/connector/src/sink/kafka.rs

```diff
@@ -134,6 +140,20 @@ impl<const APPEND_ONLY: bool> KafkaSink<APPEND_ONLY> {
         })
     }
 
+    pub async fn validate(config: KafkaConfig, pk_indices: Vec<usize>) -> Result<()> {
+        // For upsert Kafka sink, the primary key must be defined.
+        if !APPEND_ONLY && pk_indices.is_empty() {
+            return Err(SinkError::Config(anyhow!(
+                "primary key not defined for upsert kafka sink (please define in `primary_key` field)"
+            )));
+        }
+
+        // Try Kafka connection.
+        KafkaTransactionConductor::new(config).await?;
```

Comment:
can we reuse the conductor instance? it has some overhead to recreate a connection to kafka broker

Reply:
May refine this in the future.